x-pack/plugins/task_manager/server/MONITORING.md

```diff
@@ -0,0 +1,242 @@
+
+The root has its own `status` which indicate the state of the system overall as infered from the `status` of the section.
+An `Error` status in any section will cause the whole system to display as `Error`.
+A `Warning` status in any section will cause the whole system to display as `Warning`.
+An `OK` status will only be displayed when all sections are marked as `OK`.
+
+The root `timestamp` is the time in which the summary was exposed (either to the DEBUG logger or the http api) and the `lastUpdate` is the last time any one of the sections was updated.
+
+#### The Configuration Section
+The `configuration` section summarizes Task Manager's current configuration, including dynamic configurations which change over time, such as `poll_interval` and `max_workers` which adjust in reaction to changing load on the system.
+
+These are "Hot" stats which are updated whenever a change happens in the configuration.
+
+#### The Workload Section
+The `workload` which summarizes the work load in the current deployment, listing the tasks in the system, their types and what their current status is.
+
+It includes three sub sections:
+  - The number of tasks scheduled in the system, broken down by type and status.
+  - The number of idle `overdue` tasks, whose `runAt` has expired.
+  - Execution density in the next minute or so (configurable), which shows how many tasks are scheduled to execute in the scope of each polling interval. This can give us an idea of how much load there is on the current Kibana deployment.
+
+These are "Cold" stat which are updated at a regular cadence, configured by the `monitored_aggregated_stats_refresh_rate` config.
+
+#### The Runtime Section
+The `runtime` tracks Task Manager's performance as it runs, making note of task execution time, _drift_ etc.
+These include:
+  - The time it takes a task to run (mean and median, using a configurable running average window, `50` by default)
+  - The average _drift_ that tasks experience (mean and median, using the same configurable running average window as above). Drift tells us how long after a task's scheduled a task typically executes.
+  - The polling rate (the timestamp of the last time a polling cycle completed) and the result [`No tasks | Filled task pool | Unexpectedly ran out of workers`] frequency the past 50 polling cycles (using the same window size as the one used for running averages)
+  - The `Success | Retry | Failure ratio` by task type. This is different than the workload stats which tell you what's in the queue, but ca't keep track of retries and of non recurring tasks as they're wiped off the index when completed.
+
+These are "Hot" stats which are updated reactively as Tasks are executed and interacted with.
+
+### Example Stats
+
+For example, if you _curl_ the `/api/task_manager/_health` endpoint, you might get these stats:
```

Comment:
I keep forgetting this but we're supposed to have our API responses in snake_case based on a past [issue](https://github.com/elastic/kibana/issues/59799) and the styleguide recommendation: https://github.com/elastic/kibana/blob/master/STYLEGUIDE.md#api-endpoints.